libsolidity/codegen/MultiUseYulFunctionCollector.cpp

```diff
@@ -54,3 +56,31 @@ string MultiUseYulFunctionCollector::createFunction(string const& _name, functio
 	}
 	return _name;
 }
+
+string MultiUseYulFunctionCollector::createFunction(
+	string const& _name,
+	function<void(string&, vector<string>&, vector<string>&)> const& _creator
+)
+{
+	solAssert(!_name.empty(), "");
+	if (!m_requestedFunctions.count(_name))
+	{
+		m_requestedFunctions[_name] = "<<STUB<<";
+		vector<string> args;
+		vector<string> retValues;
+		string body;
+		_creator(body, args, retValues);
```

Comment:
Actually - sorry about another comment - we could at least have `body` be a return value, couldn't we?

Reply:
I tried adding structure containing `name`, `params`, `returnValues`, and `body`. Instead of passing `_creator` I tried passing that structure to `createFunction`

@chriseth Does it look better, is it better to do like this, or to proceed with `_creator` function?
Once we finalize the interface here, I can use it in more places instead of old interface. For now, I just update one util  function so it is easier to review

Reply:
I think we do need the `_creator` function unfortunately. See https://github.com/ethereum/solidity/pull/11066#discussion_r594160380

Reply:
Indeed...
Returned back to `_creator`